### DISCUSSION TITLE
upload to entity with fileInfo

### DISCUSSION THREAD

**Discussion by cbcdev** [strapi/strapi#6230] [Legacy Topics]
Hey all,

I can't seem to find out how to upload to a media data type on a custom model, whilst passing fileInfo for the media library/upload plugin.

So for example, if I POST directly to /upload/ with a filebuffer and a body member named "fileInfo":{"name":"myName","caption":"myCaption"} this works just fine.  If I try to upload to a media member on a custom content-type (say the field name is image) I do something like this using needle:
```

PostBody {
    data: {"field1":"value"},
    "files.image":{
        buffer: this.imgRaw, 
        content_type: "image/jpeg" 
    }
}

```

No place I put fileInfo seems to be passed along to the upload plugin.  In checking the code, it doesn't seem it does this.

So is the recommended way to do this upload directly to  upload/ and then just relate to my custom-type field?  Seems strange to offer the functionality to upload directly via an entity with no ability to pass along fileInfo for upload plugin?

But I might just be missing the solution.  Any ideas?

My secondary problem is PUTting to the image later uploads another file rather than doing replace media.  Intentional or NYI?  Or perhaps I am doing that wrong too =D

I love the new media library module but there's not much for docs on the innards.

Thanks!

EDIT: Whoops messed my code up.  Should be fixed now.

**Comment by lauriejim** [comment]
Hello, did you found your solution?

**Reply by cbcdev** [reply]
I did not.  I can see from strapi debug that itself does this in one POST in the admin panel / media library.  However, I cannot figure the data structure to do so.  I don't know of a strapi setting to enable post body output so I can see, nor am I familiar enough with the source of strapi to even find where that might be to check the code.

Right now, I do 2 separate calls -- one to upload, and a followup POST to alter the fileInfo immediately after.  It works, but I think it wouldn't be necessary if I knew the right pattern.

Thanks!

**Comment by lauriejim** [comment]
Did you check this documentation here - https://strapi.io/documentation/3.0.0-beta.x/plugins/upload.html#upload-files

**Reply by derrickmehaffy** [reply]
I think these docs need to be updated for the new meta-data provided by the media library @soupette ?
